docs/source/tutorial-pyfunc.rst

```diff
@@ -0,0 +1,368 @@
+
+- Set the ``MLFLOW_TRACKING_URI`` environment variable to the URI of the server. 
+
+.. important:: 
+    You **must** set the ``MLFLOW_TRACKING_URI`` environment variable to the URI of the server. If MLflow sees that the tracking URI is not an HTTP or HTTPS URI, it will not allow you to save or load models.
+
+Train a Model
+-------------
+
+For this tutorial, we will train a logistic regression model using scikit-learn. The task is to classify the species of an iris flower given the flower's measurements. 
+
+Create a file called **train.py** and add the following code:
+
+.. code-block:: python
+
+    from mlflow.models import ModelSignature
+    from sklearn.datasets import load_iris
+    from sklearn.linear_model import LogisticRegression
+    from sklearn.metrics import accuracy_score
+
+    import pandas as pd
+    import joblib as jl
+    import numpy as np
+
+    import mlflow
+
+    from custom_code import iris_classes
+
+
+    def build_and_train_model(X : pd.DataFrame, y : pd.Series) -> LogisticRegression:
+        classifier = LogisticRegression(C=1.0, random_state=42).fit(X, y)
+        return classifier
+
+
+    def predict_and_track(classifier : LogisticRegression, X : pd.DataFrame, y : pd.Series) -> float:
+        predictions = classifier.predict(X)
+        score = accuracy_score(y, predictions)
+        return score
+
+
+    def initialize_and_train() -> (LogisticRegression, ModelSignature):
+        X, y = load_iris(return_X_y=True, as_frame=True)
+        classifier = build_and_train_model(X, y)
+        score = predict_and_track(classifier, X, y)
+        mlflow.log_metric("accuracy", score)
+        signature = mlflow.models.infer_signature(X, y)
+        return classifier, signature
+
+
+        # Entry point
+        if __name__ == "__main__":
+            with mlflow.start_run() as run:
+                classifier, signature = initialize_and_train()
+
+Once you've created the file, run it with:
+
+.. code-block:: bash
+
+    python train.py
+
+.. note:: You may receive a warning that the solver failed to converge. However, if you check the accuracy of ``predictions`` vs ``y``, you will see that the model is highly accurate.
+
+The code:
+
+1. Imports necessary modules from ``sklearn`` and ``mlflow``. 
+2. Starts an MLflow run (``with mlflow.start_run() as run``)
+3. Within ``initialize_and_train()``: loads the iris dataset 
+4. Specifies and fits a ``LogisticRegression`` model (``build_and_train_model``)
+5. Calculates the accuracy of the model and logs it to MLflow (``predict_and_track``)
+6. Infers the signature of the model using :py:func:`mlflow.models.infer_signature` function and the input data and labels
+7. Returns the model and the signature of the model
+
+What it doesn't do is log the model! Because there is an sklearn flavor for MLflow, we could do so by adding a single line of code to main:
```

Comment:
(also, turning on autologging will log this model without needing to explicitly start a run)